pkg/util/fsync/fsync.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package fsync
+
+import (
+	"os"
+	"path"
+
+	"github.com/grafana/dskit/multierror"
+)
+
+// Run execute writeFunc to write data into passed file and then execute
+// fsync operation to make sure the file and its content are stored atomically.
+func Run(file *os.File, writeFunc func(f *os.File) (int, error)) error {
```

Comment:
`Run` doesn't quite convey the intention of this method to me. I'd suggest renaming this to something like `CreateFile`, then consumers will call something like `atomicfs.CreateFile` which conveys the intention a lot more clearly to me.

Reply:
Initially it was named `CreateFile`. But since we need to pass a created fine inside the function, `CreateFile` also doesn't accurate. Let me think a better name.